### PR TITLE
Array

### PR DESCRIPTION
**Pull request by hklerer**

### CODE REVIEW

**Comment by hklerer** [comment]
only one of my tests are passing. I was wondering if you can help me with this.

File: src/main/java/klerer/fallingsand/Sand.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    /**
+     * sets the value in field to 1
+     */
+    public void put(int x, int y) {
+        field[y][x] = 1;
+    }
+
+    public void fall() {
+        // moves all sand down one square
+        if (get(x, y) == field[1][2]) {
```
Comment (andrewoid) [inline src/main/java/klerer/fallingsand/Sand.java]: What are the values of x and y?

Comment (hklerer) [reply src/main/java/klerer/fallingsand/Sand.java]: the values in the array

Comment (andrewoid) [reply src/main/java/klerer/fallingsand/Sand.java]: No. That's not the way variables work...

Comment (hklerer) [reply src/main/java/klerer/fallingsand/Sand.java]: do i need to pass them in as parameters or I'm supposed to do it a different way?

Comment (andrewoid) [reply src/main/java/klerer/fallingsand/Sand.java]: The comment says, "move the sand down one square"
The fall() method takes no arguments.

You are supposed to have an idea how to do that.

File: src/main/java/klerer/fallingsand/Sand.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    public void fall() {
+        // moves all sand down one square
+        if (field[1][2] == 1) {
+            field[1][3] = 1;
+        } else if (field[1][0] == 1) {
+            field[1][0] = 0;
+            field[1][1] = 1;
+        }
```
Comment (andrewoid) [inline src/main/java/klerer/fallingsand/Sand.java]: This code might pass the tests. But if I tell you that field needs to be 100x100 and sand can appear anywhere then this isn't going to work.

File: src/main/java/klerer/fallingsand/Sand.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    public void fall() {
+        // moves all sand down one square
+        if (field[1][2] == 1) {
+            field[intY][intX + 1] = 1;
+        } else if (field[1][0] == 1) {
+            field[1][0] = 0;
+            field[intY][intX + 1] = 1;
+        }
```
Comment (andrewoid) [inline src/main/java/klerer/fallingsand/Sand.java]: You need to evaluate ALL the elements in the array. Not just the ones used in the tests.

**Review by andrewoid — APPROVED** [review]

**Pull request merged**